### PR TITLE
GeoDns improvements

### PR DESCRIPTION
**Pull request by sjcorbett**
A variety of improvements to the geo-DNS entities. In particular:
* Entities whose service state is not `RUNNING` are automatically excluded from the pool of targets. This can be disabled to maintain existing behaviour.
* The group to be tracked by `AbstractGeoDnsService` can be given in a config key, which means users don't need to call `setTargetEntityProvider` and thus that the service can be used from yaml.

### CODE REVIEW

File: software/webapp/src/test/java/org/apache/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java [hunk 1]
```diff
@@ -100,12 +106,9 @@
 ... 31 more
  */
 @Test(groups="Broken", enabled=false)
```
Comment (ahgittin) [inline software/webapp/src/test/java/org/apache/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: still broken?

Comment (sjcorbett) [inline software/webapp/src/test/java/org/apache/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: Yes, for reasons described here: https://issues.apache.org/jira/browse/BROOKLYN-200

File: usage/camp/src/test/resources/org/apache/brooklyn/camp/brooklyn/geodns.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# specific language governing permissions and limitations
+# under the License.
+#
+services:
+
+- name: Web cluster
+  type: org.apache.brooklyn.entity.group.DynamicRegionsFabric
+  id: web-fabric
+
+  # Location required but test should not do any provisioning.
```
Comment (ahgittin) [inline usage/camp/src/test/resources/org/apache/brooklyn/camp/brooklyn/geodns.yaml]: would be nice if locations weren't required -- but not needed for this PR

**Comment by ahgittin** [comment]
looks good.  minor comments, can be addressed subsequently.  merging.

**Pull request merged**